### PR TITLE
[ABANDONED] MLeap Flavor/Deployment #2: Support for multiple Spark model flavors

### PR DESCRIPTION
**Pull request by dbczumar**
This is the **second** PR in a multi-part series that will introduce support for the MLeap flavor in MLFlow. 

This PR moves serialization logic specific to Spark MLLib (SparkML) into a separate `sparkml.py` module. In a following PR, we will add an `mleap.py` module. The existing `spark.py` module will still be used as the entrypoint for serializing PySpark models; it will leverage `sparkml.py` and `mleap.py` to output supported flavors at serialization time.

For context about how these changes fit into the overall MLeap deployment integration, see #278

### CODE REVIEW

File: mlflow/sparkml.py [hunk 1]
```diff
@@ -0,0 +1,172 @@
+"""
```
Comment (dbczumar) [inline mlflow/sparkml.py]: @tomasatdatabricks Is `mllib.py` a better name for this module? @aarondav Pointed out that the current name is very similar to `spark.py`

**Review by tomasatdatabricks — CHANGES_REQUESTED** [review]
Looks good, thanks Corey!

I left a few comments, most of them minor. My only concerns is that I don't like how we handle the model save. I think we should catch exceptions, we can let users specify which flavors they want and fail the method if any of the requested flavors fails to save.

File: mlflow/sparkml.py [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+    if not isinstance(spark_model, Transformer):
+        raise SaveModelException("Unexpected type {}." 
+                                 " Spark MLLib serialization only works with Spark Transformers".
+                format(str(type(spark_model))))
+    if not isinstance(spark_model, PipelineModel):
+        raise SaveModelException("Not a PipelineModel." 
+                                 " Spark MLLib can currently only save PipelineModels.")
+
+    spark_path_full = os.path.join(path, "sparkml")
+    sparkml_datapath_sub = os.path.join("sparkml", "model")
```
[the remark this answers is not in the document]
Comment (tomasatdatabricks) [reply mlflow/sparkml.py]: Oh and I would prefer if sparkml.add_to_model would be called save_model, cause that's more consistent with how other things are named.

**Comment by dbczumar** [comment]
After discussing the structure of this PR with @tomasatdatabricks, we agree that preserving `spark.py` as the MLLib-specific point of entry makes more sense. `spark.py` will still contain logic for MLLib pipeline serialization and call out to a separate `mleap.py` method to add the MLeap flavor, if relevant.

This is superseded by #324

**Pull request closed without merging**